rcp/org.testeditor.rcp4.views.tcltestrun.tests/src/org/testeditor/rcp4/views/tcltestrun/rest/TestExecutionLogServiceTest.xtend

```diff
@@ -72,25 +73,56 @@ class TestExecutionLogServiceTest extends AbstractTest {
 		// given
 		testExecLogService.testExecutionManager = executionManager
 		val teLog = new TestExecutionLog
-		teLog.logFile = tempFolder.newFile("te-2016.11.16.-22:24.log")
-		Files.write(teLog.logFile.toPath, "Log content".bytes)
+		val baseDir = tempFolder.newFolder("testrun-2016.11.16-22:24")
+		teLog.logDir = new File(baseDir,"testrun.log")
+		Files.write(teLog.getLogDir.toPath, "Log content".bytes)
+		teLog.logDir = new File(baseDir,"testSummary.xml")
+		Files.write(teLog.getLogDir.toPath, TestRunUtility.testResult.bytes)
 		when(logLocationHelper.logLocation).thenReturn(tempFolder.root)
 
 		// when
 		val listString = testExecLogService.testLogExeutionsList.entity as String
 		val gson = new Gson
 		val json = gson.fromJson(new StringReader(listString), TestExecutionLogList)
 		val links = json.entries.get(0).links
-		val logString = testExecLogService.getTestLogExeutionContent("te-2016.11.16.-22:24.log").entity as String
+		val logString = testExecLogService.getTestLogExeutionContent("testrun-2016.11.16-22:24").entity as String
 		val log = gson.fromJson(new StringReader(logString), TestExecutionLog)
 
 		// then
 		assertEquals(links.length, 3)
 		assertEquals(links.get(0).href,
-			TestExecutionLogService.SERVICE_PATH + "/te-2016.11.16.-22:24.log/fullLogs")
+			TestExecutionLogService.SERVICE_PATH + "/testrun-2016.11.16-22:24/fullLogs")
 		assertEquals(links.get(1).href,
-			TestExecutionLogService.SERVICE_PATH + "/te-2016.11.16.-22:24.log/logGroups")
+			TestExecutionLogService.SERVICE_PATH + "/testrun-2016.11.16-22:24/logGroups")
 		log.content.assertEquals("Log content")
 	}
 
+	@Test
+	def void testGetTestLogExeutionTestStepTree() {
+		// given
+		testExecLogService.testExecutionManager = executionManager
+		val teLog = new TestExecutionLog
+		val baseDir = tempFolder.newFolder("testrun-2016.11.16-22:24")
+		teLog.logDir = new File(baseDir,"testrun.log")
+		val logString = '''
+			[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---
+			18:49:10 INFO  [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Test specification] * Given
+			18:49:10 TRACE [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Component] TestEditorServices		
+		'''
+		Files.write(teLog.logDir.toPath, logString.bytes)
+		teLog.logDir = new File(baseDir,"testSummary.xml")
+		Files.write(teLog.getLogDir.toPath, TestRunUtility.testResult.bytes)
+		when(logLocationHelper.logLocation).thenReturn(tempFolder.root)
+		
+		// when
+		val response = testExecLogService.getTestLogExeutionTestStepTree("testrun-2016.11.16-22:24")
+		val gson = new Gson
+		val log = gson.fromJson(new StringReader(response.entity as String), TestExecutionLog)
```

Comment:
Repeated code that should be extracted into a separate method (same as in `testFullLogsFromListItem`).